lib/App/DuckPAN/Perl.pm

```diff
@@ -109,54 +111,73 @@ sub duckpan_install {
 		$sp =~ s/\:\:/_/g;
 
 		# special case: check for a pinned verison number
-		my $pin_version            = $ENV{$sp};
-		my $localver               = $self->get_local_version($package);
-		my $duckpan_module_version = version->parse($module->version);
-		my $duckpan_module_url     = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
+		my $pinned_version      = $ENV{$sp};
+		my $installed_version   = $self->get_local_version($package);
+		my $latest_version      = version->parse($module->version);
+		my $duckpan_module_url  = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
 
-		$localver ||= 1e-6 if ($pin_version); # a silly, but true, value if missing and we need to compare with pinned.
```

Comment:
Ok, again, that's a special case.  In that case it *should* say you have a newer version and leave it alone.  What behavior are you expecting?  All of these are internal development edge cases imo.  If you're dzil installing you're going under the hood and should be able get out of it, e.g. use `cpanm -U` first on you non-existent version and then `duckpan upgrade`.

Like I said, we could do `duckpan latest`.  In this case it would be like a reinstall but it would try to install `pinned` or `latest` versions, ignoring whatever mess created during development, e.g. installed_version.